packages/firestore/src/local/indexeddb_remote_document_cache.ts

```diff
@@ -444,6 +375,29 @@ export class IndexedDbRemoteDocumentCache implements RemoteDocumentCache {
     return documentGlobalStore(txn).put(DbRemoteDocumentGlobal.key, metadata);
   }
 
+  /**
+   * Decodes `remoteDoc` and returns the document (or null, if the document
+   * corresponds to the format used for sentinel deletes).
+   */
+  private maybeDecodeDocument(
+    dbRemoteDoc: DbRemoteDocument | null
+  ): MaybeDocument | null {
+    if (dbRemoteDoc) {
+      const doc = this.serializer.fromDbRemoteDocument(dbRemoteDoc);
+      if (
+        doc instanceof NoDocument &&
+        doc.version.isEqual(SnapshotVersion.forDeletedDoc())
+      ) {
+        // The document is a sentinel delete and should only be used in the
+        // change log.
```

Comment:
Since we don't have a proper "change log" anymore, perhaps change to "used in getNewDocumentChanges()" ?